### PR TITLE
#24: idx check

### PR DESCRIPTION
**Pull request by olenagerasimova**
For #24 added idx check into `RequestLineFrom#part` and some tests.

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#119](https://github.com/artipie/http/issues/119) is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/5b54c001-d511-4e38-a4eb-f1e7e47ae5fb); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/main/java/com/artipie/http/rq/RequestLineFrom.java [hunk 1]
```diff
@@ -43,6 +43,12 @@
  */
 public final class RequestLineFrom {
 
+    /**
+     * Valid HTTP request line must contains 3 parts which can be splitted by
+     * whitespace char.
+     */
+    private static final int PARTS_AMOUNT = 3;
```
Comment (Vatavuk) [inline src/main/java/com/artipie/http/rq/RequestLineFrom.java:50]: @olenagerasimova this doesn't have to be a static variable, we can hardcode `3` in the `part` method

File: src/test/java/com/artipie/http/rq/RequestLineFromTest.java [hunk 2]
```diff
@@ -39,42 +41,68 @@
...
+                () -> new RequestLineFrom(
+                    String.format("%s /wallet/or/life HTTP/1.1\n", method)
+                ).method()
+            ).getMessage(),
+            new IsEqual<>(String.format("Unknown method: '%s'", method))
+        );
+    }
+
+    @Test
+    void throwsExceptionIfLineIsInvalid() {
```
Comment (Vatavuk) [inline src/test/java/com/artipie/http/rq/RequestLineFromTest.java:98]: @olenagerasimova let's create one test with 4 parts in http line

**Comment by olenagerasimova** [comment]
@Vatavuk thanks, corrected

File: src/main/java/com/artipie/http/rq/RequestLineFrom.java [hunk 3]
```diff
@@ -99,8 +93,9 @@ public String version() {
      * @return Part string
      */
     private String part(final int idx) {
+        final int pcount = 3;
         final String[] parts = this.line.trim().split("\\s");
-        if (parts.length == RequestLineFrom.PARTS_AMOUNT) {
+        if (parts.length == pcount) {
```
Comment (Vatavuk) [inline src/main/java/com/artipie/http/rq/RequestLineFrom.java:98]: @olenagerasimova I believe its a bit more easier to digest `parts.length == 3` than having a separate variable.

Comment (olenagerasimova) [reply src/main/java/com/artipie/http/rq/RequestLineFrom.java:98]: @Vatavuk it's a magic number, that's why I initially extract in into constant: unlike variables constants are properly documented with the answer to the questions "where the hell this number came from"...

Comment (Vatavuk) [reply src/main/java/com/artipie/http/rq/RequestLineFrom.java:98]: @olenagerasimova ok let's then expand the description for `part` method explaining that http line must contain 3 parts.

**Comment by olenagerasimova** [comment]
@Vatavuk please take another look

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @vatavuk[/z](https://www.0crat.com/u/vatavuk), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
Job was finished in 23 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by sereshqua** [comment]
@0crat quality good

**Comment by sereshqua** [comment]
@0crat quality good

**Comment by Vatavuk** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/artipie/http/issues/119#issuecomment-615451504))

@Vatavuk This is what I know about this job in CT2E6TK9B, as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#119](https://github.com/artipie/http/issues/119) is not in scope
  * Job [footprint](https://www.0crat.com/footprint/CT2E6TK9B?q=%7Bjob%3A%27gh%3Aartipie%2Fhttp%23119%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by 0crat** [comment]
> @0crat quality good ([here](https://github.com/artipie/http/issues/119#issuecomment-615333448))

@sereshqua There is no quality review for [#119](https://github.com/artipie/http/issues/119), no performer

**Pull request merged**